src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1291,6 +1416,78 @@ def construct_payload(self):
+
+        ingress = self.get_argument_ingress()
+        target_port = self.get_argument_target_port()
+        dockerfile = "Dockerfile"
+
+        # Parse resource group name and managed env name
+        env_id = self.containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image() is None else _reformat_image(source=self.get_argument_source(), image=self.get_argument_image(), repo=None)
+        location = self.containerapp_def["location"]
+
+        hasDockerfile = _has_dockerfile(self.get_argument_source(), dockerfile)
+        if hasDockerfile:
+            dockerfile_content = _get_dockerfile_content(repo=None, branch=None, token=None, source=self.get_argument_source(), context_path=None, dockerfile=dockerfile)
+            ingress, target_port = _get_ingress_and_target_port(self.get_argument_ingress(), self.get_argument_target_port(), dockerfile_content)
+
+        # Construct ContainerApp
+        resource_group = ResourceGroup(cmd, env_rg, location=location)
+        env = ContainerAppEnvironment(cmd, env_name, resource_group, location=location)
+        app = ContainerApp(cmd=cmd, name=self.get_argument_name(), resource_group=resource_group, image=image, env=env, target_port=target_port, workload_profile_name=self.get_argument_workload_profile_name(), ingress=ingress, registry_server=registry_server, registry_user=registry_user, registry_pass=registry_pass)
+
+        # Fetch registry credentials
+        _get_registry_details(cmd, app, self.get_argument_source())  # fetch ACR creds from arguments registry arguments
+
+        # Uses buildpacks to generate image if Dockerfile was not provided by the user
+        app.run_acr_build(dockerfile, self.get_argument_source(), quiet=False, build_from_source=not hasDockerfile)
+
+        # Validate an image associated with the container app exists
+        if self.containerapp_def["properties"]["template"]["containers"] is None or len(self.containerapp_def["properties"]["template"]["containers"]) == 0:
+            raise ValidationError(
+                "Error: The containerapp '{}' does not have containers associated with it. Please specify an image using the --image argument while creating the ContainerApp".format(
+                    self.get_argument_name()))
+        self.new_containerapp["properties"]["template"] = {} if "template" not in self.new_containerapp["properties"] else self.new_containerapp["properties"]["template"]
+        self.new_containerapp["properties"]["template"]["containers"] = self.containerapp_def["properties"]["template"]["containers"]
+        if len(self.new_containerapp["properties"]["template"]["containers"]) == 1:
```

Comment:
It is probably not necessary to check if multiple sidecar containers exist or not because the registry server is extracted from the [first container](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/containerapp_decorator.py#L1429) so it may be okay to just update ` self.new_containerapp["properties"]["template"]["containers"][0]["image"]`